src/ILCompiler.Compiler/src/CppCodeGen/CppWriter.cs

```diff
@@ -813,15 +875,18 @@ private String GetCodeForType(TypeDesc type)
             if (type.IsString)
             {
                 // String has non-standard layout
-                // component size = 2, flags = 0, base size = 2 ptrs + dword + first char
-                sb.Append("{ sizeof(uint16_t), 0, 2 * sizeof(void*) + sizeof(int32_t) + 2, ");
+                sb.Append("{ sizeof(uint16_t), 0x"); // component size
+                sb.Append(flags.ToString("x", CultureInfo.InvariantCulture));
+                sb.Append(", 2 * sizeof(void*) + sizeof(int32_t) + 2, "); // base size
```

Comment:
It does not really matter - they can be used interchangeably. I think void* makes more sense in this context.